app/src/main/java/team7/voluntime/Fragments/Common/UserProfileFragment.java

```diff
@@ -1,42 +1,148 @@
+    TextView uPhone;
+
+    @BindView(R.id.uAddress)
+    TextView uAddress;
+
+    @BindView(R.id.uGen)
+    TextView uGen;
+
+    @BindView(R.id.uDob)
+    TextView uDob;
 
 
     public UserProfileFragment() {
         // Required empty public constructor
     }
 
-
     @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+
         // Inflate the layout for this fragment
         View v = inflater.inflate(R.layout.fragment_user_profile, container, false);
         ButterKnife.bind(this, v);
+
+        //Get Current logedin instance from database
+        mAuth = FirebaseAuth.getInstance();
+        mUser = mAuth.getCurrentUser();
+        //Get Accoount type and pass it into getType() method to return corects string path
+        DatabaseReference typeRef = FirebaseDatabase.getInstance().getReference(getType());
+
+        typeRef.child(mUser.getUid()).addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                for (DataSnapshot snap : dataSnapshot.getChildren()) {
+                    if (getType().equals("Volunteers")) {
+                        volunteerStuff.setVisibility(VISIBLE);
+                        Type = dataSnapshot.child("AccountType").getValue().toString();
+                        UserName = dataSnapshot.child("Profile").child("FullName").getValue().toString();
+                        Phone = dataSnapshot.child("Profile").child("PhoneNumber").getValue().toString();
+                        Address = dataSnapshot.child("Profile").child("Address").getValue().toString();
+
+                        DOB = dataSnapshot.child("Profile").child("DateOfBirth").getValue().toString();
+                        uDob.setText(DOB);
+
+                        GEN = dataSnapshot.child("Profile").child("Gender").getValue().toString();
+                        uGen.setText(GEN);
+                    }
+                    if (getType().equals("Charities")) {
+                        charityStuff.setVisibility(VISIBLE);
+                        Type = dataSnapshot.child("AccountType").getValue().toString();
+                        UserName = dataSnapshot.child("Profile").child("Name").getValue().toString();
+                        Address = dataSnapshot.child("Profile").child("Address").getValue().toString();
+                        Phone = dataSnapshot.child("Profile").child("PhoneNumber").getValue().toString();
+                        Category = dataSnapshot.child("Profile").child("Category").getValue().toString();
+                        uCat.setText(Category);
+                        Desc = dataSnapshot.child("Profile").child("Description").getValue().toString();
+                        uDesc.setText(Desc);
+                    }
```

Comment:
So, if you see my recent PR or what was merged, we dont need to set each value to a unique variable, what we could do (and is alot cleaner), is set the values of the datasnapshot to a class. Then when we do 'setText()', we could pass in the value using a getter for each variable. If you have a look at my recent changes, all i needed to do for 'ViewEventsFragment' is: `charity = dataSnapshot.getValue(Charity.class);` and this automatically sets the values for each of the fields in the class (except for nested fields where you will need to manually set them yourself). (If you have any questions on this let me know and ill walk you through it).

Reply:
Not sure exactly how to do this, understand what you are getting at though